src/main/java/dk/sdu/se_f22/searchmodule/twowaysynonyms/TwoWaySynonym.java

```diff
@@ -79,16 +81,11 @@ public ArrayList<Synonym> readAll(int groupId) {
     public Synonym read(String synonym) {
         String statementFormat = "SELECT * FROM twoway_synonym WHERE synonym = ?";
         ArrayList<Object> args = new ArrayList<>(){{add(synonym);}};
-        PreparedStatement stmt = getPreparedStatement(statementFormat, args);
-        ResultSet rs = readDatabase(stmt);
-        try {
-            if (rs.next()) {
-                return new Synonym(
-                        rs.getString(1), rs.getString(2), rs.getInt(3)
-                );
-            }
-        } catch (SQLException ex) {
-            ex.printStackTrace();
+        Connection conn = getConnection();
+        PreparedStatement stmt = getPreparedStatement(statementFormat, args, conn);
+        ArrayList<Synonym> synonyms = readDatabase(stmt, conn);
+        if (!synonyms.isEmpty()) {
```

Comment:
Could still produce an NPE since readDatabase can still return a null value.

Reply:
The public function may return null, but there won't be any NPE inside the logic.

Reply:
Ah nvm, if exception is caught it will return null. Will fix